polyfill/lib/yearmonth.mjs

```diff
@@ -62,10 +62,11 @@ export class YearMonth {
   plus(durationLike, disambiguation = 'constrain') {
     if (!ES.IsYearMonth(this)) throw new TypeError('invalid receiver');
     disambiguation = ES.ToArithmeticDisambiguation(disambiguation);
-    const duration = ES.ToLimitedDuration(durationLike, [DAYS, HOURS, MINUTES, SECONDS, MILLISECONDS, MICROSECONDS, NANOSECONDS]);
+    const duration = ES.ToLimitedDuration(durationLike);
     let { year, month } = this;
-    const { years, months } = duration;
-    ({ year, month } = ES.AddDate(year, month, 1, years, months, 0, disambiguation));
+    const { years, months, hours, minutes, seconds, milliseconds, microseconds, nanoseconds } = duration;
+    const { days } = ES.BalanceDuration(duration.days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds, 'days');
+    ({ year, month } = ES.AddDate(year, month, 1, years, months, days, disambiguation));
```

Comment:
We forgot to discuss YearMonth this morning, but your solution here seems fine on first glance.